sql/rename.go

```diff
@@ -209,6 +213,8 @@ func (p *planner) RenameIndex(n *parser.RenameIndex) (planNode, error) {
 	if err := p.txn.Put(descKey, tableDesc); err != nil {
 		return nil, err
 	}
+	// Mark transaction as operating on the system DB.
+	p.txn.SetSystemDBTrigger()
```

Comment:
maybeGossipConfigs is currently called from two places:
* NewReplica (I call maybeGossipSystemDB here too)
* applyRaftCommand with `txn == nil && proto.IsWrite`. all of our systemDB-related writes are part of a transaction, and we can't just remove the `txn==nil` check or we'll just be consuming lots of intents. The post-commit trigger gives us the right hook.

One alternative is to have a post-commit trigger that looks at all intents on EndTransaction, but I really don't want to examine every single key being written in a transaction.
Another is to do the very same in the transaction client-side, but I really don't want to scan keys there either.

I agree the trigger setting is a bit fragile right now. However, systemdb stuff is already somewhat special (we should always call Validate before writing a descriptor). My plan is to relocate all systemDB ops into sql/descriptor (or similar) and do the proper logic there.